lib/govuk/diff/pages/visual_diff/runner.rb

```diff
@@ -0,0 +1,30 @@
+require 'yaml'
+require 'tempfile'
+require 'open-uri'
+
+module Govuk
+  module Diff
+    module Pages
+      module VisualDiff
+        class Runner
+          def initialize(list_of_pages_uri:)
```

Comment:
I'd do the YAML parsing in the rake task and provide an array of URIs to this class.